Goroutine leak in TestPeerConnection_Close_PreICE

### Your environment.
  - Version: d782e9bd4abb87c778b0bf532499e7d85a6b6b95

### What happened?

Getting this test failure intermittently on CI
```
--- FAIL: TestPeerConnection_Close_PreICE (10.21s)
    util.go:38: Unexpected routines: 
        goroutine 477 [select]:
        github.com/pion/ice.(*Agent).connect(0xc0000d6480, 0xcabce0, 0xc0000a2018, 0x0, 0xc0000d52ea, 0x10, 0xc00009e998, 0x20, 0x0, 0x0, ...)
        	/home/travis/gopath/pkg/mod/github.com/pion/ice@v0.2.1/transport.go:44 +0x255
        github.com/pion/ice.(*Agent).Accept(...)
        	/home/travis/gopath/pkg/mod/github.com/pion/ice@v0.2.1/transport.go:21
        github.com/pion/webrtc/v2.(*ICETransport).Start(0xc0000a63c0, 0xc0003bef00, 0xc0000d52ea, 0x10, 0xc00009e998, 0x20, 0x0, 0xc000269e98, 0x0, 0x0)
        	/home/travis/build/pion/webrtc/icetransport.go:117 +0x3fa
        github.com/pion/webrtc/v2.(*PeerConnection).SetRemoteDescription.func2(0xc000000000, 0xc000089080, 0xc0000d52ea, 0x10, 0xc00009e998, 0x20, 0xc00003068c, 0x7, 0xc000030694, 0x5f)
        	/home/travis/build/pion/webrtc/peerconnection.go:979 +0x1b2
        created by github.com/pion/webrtc/v2.(*PeerConnection).SetRemoteDescription
        	/home/travis/build/pion/webrtc/peerconnection.go:970 +0x169e
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,3 +1,8 @@
+language: c
+cache:
+  directories:
+    - $HOME/.cache/go-build
+    - $HOME/gopath/pkg/mod
 env:
   - GO111MODULE=on
 

--- a/peerconnection_close_test.go
+++ b/peerconnection_close_test.go
@@ -1,3 +1,5 @@
+// +build !js
+
 package webrtc
 
 import (
@@ -59,7 +61,7 @@ func TestPeerConnection_Close(t *testing.T) {
 // Assert that a PeerConnection that is shutdown before ICE starts doesn't leak
 func TestPeerConnection_Close_PreICE(t *testing.T) {
 	// Limit runtime in case of deadlocks
-	lim := test.TimeOut(time.Second * 20)
+	lim := test.TimeOut(time.Second * 10)
 	defer lim.Stop()
 
 	report := test.CheckRoutines(t)
@@ -87,5 +89,20 @@ func TestPeerConnection_Close_PreICE(t *testing.T) {
 	err = pcAnswer.Close()
 	if err != nil {
 		t.Fatal(err)
+	}
+
+	// Assert that ICEGatherer is shutdown, test timeout will prevent deadlock
+	pcAnswer.iceTransport.lock.Lock()
+	gatherer := pcAnswer.iceTransport.gatherer
+	pcAnswer.iceTransport.lock.Unlock()
+	for {
+		gatherer.lock.RLock()
+		if gatherer.agent == nil {
+			time.Sleep(time.Second * 3)
+			return
+		}
+		gatherer.lock.RUnlock()
+
+		time.Sleep(time.Second)
 	}
 }

